### DISCUSSION TITLE
Retain 'styles' in auto1111

### DISCUSSION THREAD

**Discussion by Alintya** [AbdBarho/stable-diffusion-webui-docker#377] [Ideas]
Is it intended that styles seem to get purged on updates?

**Comment by AbdBarho** [comment]
No, this is just a consequence of how symlinks works and how the webui writes to the file. https://github.com/AbdBarho/stable-diffusion-webui-docker/pull/180

Ideas welcome.

**Comment by DevilaN** [comment]
You cannot do stuff like this with bind-mounted files.

We probably need hacky-tricky way with bind mounting local file `./data/config/auto/styles.csv` to another location like `/localstuff/styles.csv` and...
1. With `startup.sh` copy `/localstuff/styles.csv` to `/stable-diffusion-webui/styles.csv` if it does not exists (clean container).
2. With another simple script running in background use inotifywait to monitor `/stable-diffusion-webui/styles.csv` file for changes and copy changes to `/localstuff/styles.csv` so bind mounted file will remain up to date with current styles.csv inside container.

**Comment by Kadah** [comment]
Looks like a solution now exists upstream.

https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/e10ee9627233a0e0d7cf97e8b582ce543b48439e

Also, --styles-file arg does exists.

https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/5eee2ac39863f9e44591b50d0710dd2615416a13

**Comment by Saheyus** [comment]
For anyone interested, the style.csv is now easily accessible in \data\config\auto (and therefore easier to save and replace in case of issue)
